### ISSUE TITLE
"Invalid Ethereum signature returned"

### ISSUE THREAD

**Issue by shrugs** [GridPlus/gridplus-sdk#499]
I'm getting "Invalid Ethereum signature returned" when trying to sign specific messages (namely, blur's collection bid signature), both with metamask and frame clients. could be related to #452 if this patch hasn't landed in either metamask or frame.

**Comment by shrugs** [comment]
i've confirmed that this still occurs in `frame@v0.5.0-canary.9`, which seems to have included the gridplus sdk version that includes the fix from #452 (https://github.com/floating/frame/pull/1000)

**Comment by shrugs** [comment]
now there's a 14-day time limit for us to solve this so i can claim my blur airdrop lmao

**Comment by alex-miller-0** [comment]
@shrugs have you tried setting a non-zero royalty? If not, this is a known bug that will be fixed in Lattice firmware v0.16.4, which is currently undergoing testing: https://github.com/GridPlus/lattice-software-releases/releases/tag/hsm-0.16.4

**Comment by shrugs** [comment]
this issue has occurred when listing NFTs before, and it _does_ correctly sign if i set the royalties to > 0 (nice!)

but this also occurs when signing a collection bid, which may or may not be the same issue:

```json
{
    "domain": {
        "name": "Blur Exchange",
        "version": "1.0",
        "chainId": 1,
        "verifyingContract": "0x000000000000ad05ccc4f10045630fb830b95127"
    },
    "types": {
        "Order": [
            {
                "name": "trader",
                "type": "address"
            },
            {
                "name": "side",
                "type": "uint8"
            },
            {
                "name": "matchingPolicy",
                "type": "address"
            },
            {
                "name": "collection",
                "type": "address"
            },
            {
                "name": "tokenId",
                "type": "uint256"
            },
            {
                "name": "amount",
                "type": "uint256"
            },
            {
                "name": "paymentToken",
                "type": "address"
            },
            {
                "name": "price",
                "type": "uint256"
            },
            {
                "name": "listingTime",
                "type": "uint256"
            },
            {
                "name": "expirationTime",
                "type": "uint256"
            },
            {
                "name": "fees",
                "type": "Fee[]"
            },
            {
                "name": "salt",
                "type": "uint256"
            },
            {
                "name": "extraParams",
                "type": "bytes"
            },
            {
                "name": "nonce",
                "type": "uint256"
            }
        ],
        "Fee": [
            {
                "name": "rate",
                "type": "uint16"
            },
            {
                "name": "recipient",
                "type": "address"
            }
        ]
    },
    "value": {
        "trader": "0x", // <- note: removed my address here for privacy
        "side": 0,
        "matchingPolicy": "0x0000000000b92d5d043faf7cecf7e2ee6aaed232",
        "collection": "0x5979e05c1c1874da71036aa7fb42c5657888ae2f",
        "tokenId": "0",
        "amount": "1",
        "paymentToken": "0x0000000000a39bb272e79075ade125fd351887ac",
        "price": "10000000000000000",
        "listingTime": 1670341874,
        "expirationTime": 1701877874,
        "fees": [],
        "salt": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020ca1456445095106cf015fb08066b33",
        "extraParams": "0x01",
        "nonce": "0"
    }
}
```

**Comment by shrugs** [comment]
ok! fingers crossed, thanks for the clarification, and I'm glad there will now be a google result for this topic as well

**Comment by quaekr** [comment]
Seeing the same issue across the board trying to sign anything.....

**Comment by alex-miller-0** [comment]
These errors should now be fixed in firmware v0.17.0, which was released this morning. Please let me know if you are still having issues.

**Comment by shrugs** [comment]
confirmed fixed! ty 🙏🫡
